frontend/src/components/dashboard/aliases/MaskCard.tsx

```diff
@@ -251,15 +258,53 @@ export const MaskCard = (props: Props) => {
                     />
                   )}
                   {l10n.getString(
-                    "profile-promo-email-blocking-option-promotionals"
+                    "profile-promo-email-blocking-option-promotions"
                   )}
                 </BlockLevelOption>
-                <BlockLevelOption value="all">
+                <BlockLevelOption
+                  value="all"
+                  setPromoSelectedState={setPromoIsSelectedState}
+                >
                   {l10n.getString("profile-promo-email-blocking-option-all")}
                 </BlockLevelOption>
               </BlockLevelSegmentedControl>
+              {promoIsSelected && !props.profile.has_premium ? (
+                <div
+                  className={styles["promotions-locked-description-wrapper"]}
+                >
+                  <strong>
+                    <LockIcon
+                      alt={l10n.getString(
+                        "profile-promo-email-blocking-description-promotionals-locked-label"
+                      )}
+                    />
+                    {l10n.getString(
+                      "profile-promo-email-blocking-description-promotionals-locked-label"
+                    )}
+                  </strong>
+                  <p>
+                    {l10n.getString(
+                      "profile-promo-email-blocking-description-promotionals"
+                    )}
+                  </p>
+                  <Link
+                    href="/premium#pricing"
+                    className={styles["upgrade-btn"]}
+                  >
+                    {l10n.getString("banner-pack-upgrade-cta")}
+                  </Link>
+                </div>
+              ) : null}
             </div>
-            <div className={styles["block-level-setting-description"]}>
+            {/* Only add chevron on mobile for premium users */}
+            <div
+              className={`${styles["block-level-setting-description"]}
+              ${
+                promoIsSelected &&
```

Comment:
Prettier has probably moved this around, so might be good to move the comment down a bit to be closer to the thing it's about.

```suggestion
            <div
              className={`${styles["block-level-setting-description"]}
              ${
                // Only add chevron on mobile for premium users
                promoIsSelected &&
```